.github/workflows/build-prod.yml

```diff
@@ -0,0 +1,51 @@
+# On merge into the `production` branch, build the image and deploy to ECR.
+# This is on `nypl` account.
+name: Publish production
+on:
+  pull_request:
+    branches: [production]
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  publish_qa:
+    name: Publish image to ECR and update ECS stack
+    # if: github.event.pull_request.merged
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v3
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          role-to-assume: arn:aws:iam::463786892988:role/GithubActionsDeployerRole
+          aws-region: us-east-1
+
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+
+      - name: Build, tag, and push image to Amazon ECR
+        env:
+          ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+          ECR_REPOSITORY: nyplglobalheader
+          IMAGE_TAG: ${{ github.sha }}
+        run: |
+          docker build -t $ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG .
+          MANIFEST=$(aws ecr batch-get-image --repository-name $ECR_REPOSITORY --image-ids imageTag="production-latest" --output json | jq --raw-output --join-output '.images[0].imageManifest')
+          aws ecr batch-delete-image --repository-name $ECR_REPOSITORY --image-ids imageTag="production-previous" || true
+          aws ecr put-image --repository-name $ECR_REPOSITORY --image-tag "production-previous" --image-manifest "$MANIFEST"
```

Comment:
@shuynh I'm most concerned about these three lines (also in the `build-qa.yml` file). 

The header app, for both QA `nypl-dev` and production `nypl` accounts, has two clusters, the "normal" stable cluster and a "previous" cluster that uses the previously working image. Greg worked on this script to get the previous working image on ECR to deploy to the "previous" ECS instances. I just copied this from the travis file.

I'd appreciate a close look at this but I imagine it might be something that we test when this is deployed to QA (as we previously test this).

Reply:
@gkallenberg can you also take a look if you have time?